sdcm/nemesis.py

```diff
@@ -1056,6 +1056,35 @@ def disrupt_modify_table(self):
         disrupt_func = getattr(self, disrupt_func_name)
         disrupt_func()
 
+    def disrupt_mgmt_backup(self):
+        self._set_current_disruption('ManagementBackup')
+        if not self.cluster.params.get('use_mgmt', default=None):
+            raise UnsupportedNemesis('Scylla-manager configuration is not defined!')
+        if not self.cluster.params.get('backup_bucket_location'):
+            raise UnsupportedNemesis('backup bucket location configuration is not defined!')
+
+        manager_node = self.monitoring_set.nodes[0]
+        manager_tool = mgmt.get_scylla_manager_tool(manager_node=manager_node)
+        self.log.debug("sctool version is : {}".format(manager_tool.version))
+
+        cluster_name = self.cluster.name
+        mgr_cluster = manager_tool.get_cluster(cluster_name)
+        if not mgr_cluster:
+            self.log.debug("Could not find cluster : {} on Manager. Adding it to Manager".format(cluster_name))
+            targets = [[n, n.ip_address] for n in self.cluster.nodes]
+            mgr_cluster = manager_tool.add_cluster(name=cluster_name, host=targets[0][1], disable_automatic_repair=True,
+                                                   auth_token=self.monitoring_set.mgmt_auth_token)
+        bucket_location_name = self.cluster.params.get('backup_bucket_location').split()
+        region = self.cluster.params.get('region_name').split()
+        for node in self.cluster.nodes:
+            update_config_file(node=node, region=region[0])
```

Comment:
What if this is already configured in the first run?
Why would it reconfigure it every time?

Reply:
the function ```update_config_file``` sets it using pyyaml package, so if it is already configured, it will "reconfigure" it, but nothing will change in the end.

Reply:
Not good, please fix.

Not only it's dangerous, but it can hide bugs with scylla-agent.
For example, memory leaks, everytime you restart the agent you may hide longevity issues.

Reply:
i don't understand... what is the relation of this ```update_config_file``` and memory leaks?